### PR TITLE
Don't fork appengine goals

### LINKED ISSUE #301: Make goals not fork
Make maven tasks not fork, so now users will be required to either
* run mvn compile appengine:run or appengine:deploy (instead of just mvn appengine:run/deploy)
* or bind the tasks to some lifecycle phase in their pom.xml

### PR DESCRIPTION
**Pull request by TadCordle**
Fixes #301.

### CODE REVIEW

File: src/test/java/com/google/cloud/tools/maven/it/AbstractSingleYamlDeployIntegrationTest.java [hunk 1]
```diff
@@ -33,6 +33,7 @@ public void testDeployStandard()
     Verifier verifier = new StandardVerifier("testDeployStandard");
 
     // execute with staging directory not present
+    verifier.executeGoal("package");
```
Comment (coollog) [inline src/test/java/com/google/cloud/tools/maven/it/AbstractSingleYamlDeployIntegrationTest.java]: Should these remove the call to the goal individually?

Comment (TadCordle) [reply src/test/java/com/google/cloud/tools/maven/it/AbstractSingleYamlDeployIntegrationTest.java]: What do you mean?

File: src/main/java/com/google/cloud/tools/maven/DeployAllMojo.java [hunk 2]
```diff
@@ -18,16 +18,14 @@
...
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 
 /**
  * Stage and deploy the application and all configs to Google App Engine standard or flexible
  * environment.
  */
-@Mojo(name = "deployAll")
-@Execute(phase = LifecyclePhase.PACKAGE)
+@Mojo(name = "deployAll", defaultPhase = LifecyclePhase.PACKAGE)
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/maven/DeployAllMojo.java]: I think we picked the wrong default when the plugin was made. DeployXYZ should probably be in the DEPLOY phase by default.

File: src/main/java/com/google/cloud/tools/maven/GenRepoInfoFileMojo.java [hunk 3]
```diff
@@ -23,14 +23,12 @@
...
 import org.apache.maven.plugin.MojoFailureException;
-import org.apache.maven.plugins.annotations.Execute;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.Parameter;
 
 /** Generates repository information files for the Stackdriver Debugger. */
-@Mojo(name = "genRepoInfoFile")
-@Execute(phase = LifecyclePhase.PREPARE_PACKAGE)
+@Mojo(name = "genRepoInfoFile", defaultPhase = LifecyclePhase.PREPARE_PACKAGE)
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/maven/GenRepoInfoFileMojo.java]: this one's fine where it is.

File: src/main/java/com/google/cloud/tools/maven/RunAsyncMojo.java [hunk 4]
```diff
@@ -18,14 +18,12 @@
...
 import com.google.cloud.tools.maven.CloudSdkAppEngineFactory.SupportedDevServerVersion;
-import org.apache.maven.plugins.annotations.Execute;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.Parameter;
 
 /** Starts running App Engine Development App Server asynchronously. */
-@Mojo(name = "start")
-@Execute(phase = LifecyclePhase.PACKAGE)
+@Mojo(name = "start", defaultPhase = LifecyclePhase.PACKAGE)
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/maven/RunAsyncMojo.java]: Similarly the devappserver tasks should probably be in DEPLOY

File: src/test/java/com/google/cloud/tools/maven/it/AbstractSingleYamlDeployIntegrationTest.java (hunk 1 above)
Comment (loosebazooka) [reply src/test/java/com/google/cloud/tools/maven/it/AbstractSingleYamlDeployIntegrationTest.java]: can you execute a list of goals?

File: src/main/java/com/google/cloud/tools/maven/GenRepoInfoFileMojo.java (hunk 3 above)
Comment (TadCordle) [reply src/main/java/com/google/cloud/tools/maven/GenRepoInfoFileMojo.java]: As in, it should fork?

Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/maven/GenRepoInfoFileMojo.java]: no, sorry, I just dind't want there to be any confusion about changing the default phase, and then I made it more confusing. Your changes here are fine.

**Comment by loosebazooka** [comment]
I think this might need a modification of the README/USER_GUiDE?

**Pull request merged**